app/src/main/kotlin/at/bitfire/davdroid/ui/PermissionsActivity.kt

```diff
@@ -104,7 +104,7 @@ class PermissionsActivity: AppCompatActivity() {
 
             // auto-reset permissions
             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
-                needKeepPermissions.value = pm.isAutoRevokeWhitelisted
+                needKeepPermissions.postValue(pm.isAutoRevokeWhitelisted)
```

Comment:
Was there a problem with direct assignment? The method is marked as `@UiThread`, so it should work.

Or if it runs on a worker thread, the annotation should be fixed.

Of course we could also replace all the `LiveData` in `PermissionsActivity.Model` by Compose MutableState if we're already at it :)

Reply:
Yes, using the same model `setValue` works as well. Right, then let's see if we can make it into a StateFlow straight away :)

Reply:
Because the model is holding UI state here, I think we can use `mutableStateOf()` directly in the model and then even don't need a flow.

Reply:
mmh, fair. We have not had the meeting on what're state of the art coding standards for DAVx5 yet, but if you'd prefer that :)